### PR TITLE
Update order and account dropins integration

### PR DESCRIPTION
**Pull request by Ferri0**
In this PR we've updated order and account dropins to latest versions and integrated following containers to boilerplate:
- `CustomerInformation` - user account dashboard
- `OrderCostSummary` - order details page
- `OrderProductsList` - order details page
- `OrderReturns` - order details page
- `ReturnsList` - standalone page un user account

**Test URLs:**
- Before: https://develop--aem-boilerplate-commerce--hlxsites.aem.live/
- After: https://order-and-account--aem-boilerplate-commerce--hlxsites.aem.live/

**Publish following documents in Sharepoint before merging:**
- [x] Publish [/customer/account.docx](https://adobe.sharepoint.com/:w:/r/sites/HelixProjects/_layouts/15/Doc.aspx?sourcedoc=%7BAF22A452-765B-4144-AB4F-BC6F71A31FEE%7D&file=account.docx&action=default&mobileredirect=true)
- [x] Publish [/customer/order.docx](https://adobe.sharepoint.com/:w:/r/sites/HelixProjects/_layouts/15/Doc.aspx?sourcedoc=%7B4A4FF9E5-AB7B-4512-B2B1-FBD7BC80A745%7D&file=orders.docx&action=default&mobileredirect=true)
- [x] Publish [/customer/sidebar-fragment.docx](https://adobe.sharepoint.com/:w:/r/sites/HelixProjects/_layouts/15/Doc.aspx?sourcedoc=%7BAE80D836-4C42-4F4B-A2E6-ABC0BF89DC12%7D&file=sidebar-fragment.docx&action=default&mobileredirect=true)
- [x] Publish[ /customer/returns.docx](https://adobe.sharepoint.com/:w:/r/sites/HelixProjects/_layouts/15/Doc.aspx?sourcedoc=%7B7002DBCA-F726-4843-8C6B-11CB1D2FE09E%7D&file=returns.docx&action=default&mobileredirect=true)
- [x] Publish [/customer/return-details.docx](https://adobe.sharepoint.com/:w:/r/sites/HelixProjects/_layouts/15/Doc.aspx?sourcedoc=%7BE4793126-CAA1-4318-9D4B-571F1DA13991%7D&file=return-details.docx&action=default&mobileredirect=true)
- [x] Publish [/customer/order-details.docx](https://adobe.sharepoint.com/:w:/r/sites/HelixProjects/_layouts/15/Doc.aspx?sourcedoc=%7B20E734B1-BC0E-45CA-9C02-0D945A5D65F2%7D&file=order-details.docx&action=default&mobileredirect=true)
- [x] Publish [/order-details.docx](https://adobe.sharepoint.com/:w:/r/sites/HelixProjects/_layouts/15/Doc.aspx?sourcedoc=%7BF4080433-3F5F-44DC-93EF-A2BC2B6D4F60%7D&file=order-details.docx&action=default&mobileredirect=true)
- [x] Publish [/return-details.docx](https://adobe.sharepoint.com/:w:/r/sites/HelixProjects/_layouts/15/Doc.aspx?sourcedoc=%7B262E7197-0632-45ED-ACC9-2E334A06A33B%7D&file=return-details.docx&action=default&mobileredirect=true)
- [x] Publish [/customer/create-return.docx](https://adobe.sharepoint.com/:w:/r/sites/HelixProjects/_layouts/15/Doc.aspx?sourcedoc=%7B78FF57C8-A471-4C34-9093-5D8C36E4802B%7D&file=create-return.docx&action=default&mobileredirect=true)
- [x] Publish [/create-return.docx](https://adobe.sharepoint.com/:w:/r/sites/HelixProjects/_layouts/15/Doc.aspx?sourcedoc=%7B4CAEE4D9-44F7-4B93-B4D9-CBDB0EB33E45%7D&file=create-return.docx&action=default&mobileredirect=true)
- [x] Publish [/placeholders.xlsx](https://adobe.sharepoint.com/:x:/r/sites/HelixProjects/_layouts/15/Doc.aspx?sourcedoc=%7BEBB6F538-9A48-4CA9-A144-5E4369A444F0%7D&file=placeholders.xlsx&action=default&mobileredirect=true)

### CODE REVIEW

**Review by loginesta — APPROVED** [review]

File: blocks/commerce-customer-information/commerce-customer-information.css [hunk 1]
```diff
@@ -0,0 +1 @@
+/* stylelint-disable no-empty-source */
```
Comment (sivaschenko) [inline blocks/commerce-customer-information/commerce-customer-information.css:1]: Are these empty CSS files necessary?

File: blocks/commerce-return-header/commerce-return-header.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
+/* eslint-disable import/no-unresolved */
+/* eslint-disable import/no-extraneous-dependencies */
+import { events } from '@dropins/tools/event-bus.js';
+import { Header, provider as uiProvider } from '@dropins/tools/components.js';
+import { CUSTOMER_RETURN_DETAILS_PATH, CUSTOMER_RETURNS_PATH } from '../../scripts/constants.js';
+
+// TODO - Will be refactored during implementation of return-details page
```
Comment (sivaschenko) [inline blocks/commerce-return-header/commerce-return-header.js:7]: I'd reference a ticket number in the TODOs that are going to be merged to ensure this is not forgotten later.

**Review by sivaschenko — COMMENTED** [review]
A couple of comments

File: blocks/commerce-customer-information/commerce-customer-information.css (hunk 1 above)
Comment (Ferri0) [reply blocks/commerce-customer-information/commerce-customer-information.css:1]: I would likely say yes, they are necessary. The CSS for blocks is imported automatically, so if we remove those empty files, we would constantly see 404 errors in the console, which could be confusing.

Screenshot 2024-11-04 at 14 46 31

File: blocks/commerce-return-header/commerce-return-header.js (hunk 2 above)
Comment (Ferri0) [reply blocks/commerce-return-header/commerce-return-header.js]: Agree 👍 
Added ticket links.

**Review by sivaschenko — APPROVED** [review]
Thanks for addressing my comments

File: blocks/commerce-return-header/commerce-return-header.js [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+  block.innerHTML = '';
+
+  const headerContainer = document.createElement('div');
+  await uiProvider.render(Header, { title: 'Return' })(headerContainer);
+
+  if (window.location.href.includes(CUSTOMER_RETURN_DETAILS_PATH)) {
+    const link = document.createElement('a');
+
+    // TODO: Add i18n when appropriate functionality will be introduced in boilerplate
+    link.innerText = '< Back to all returns';
```
Comment (sirugh) [inline blocks/commerce-return-header/commerce-return-header.js:21]: We already have [placeholders](https://github.com/hlxsites/aem-boilerplate-commerce/blob/main/blocks/product-details/product-details.js#L141) in use by PDP. Probably best to put all copy we add into this placeholders file for now.

File: scripts/__dropins__/storefront-account/api/getCustomer/graphql/getCustomer.graphql.d.ts [hunk 4]
```diff
@@ -1,2 +1,2 @@
-export declare const GET_CUSTOMER = "\n  query GET_CUSTOMER {\n    customer {\n     firstname\n     lastname\n     email\n    }\n  }\n";
+export declare const GET_CUSTOMER: string;
```
Comment (sirugh) [inline scripts/__dropins__/storefront-account/api/getCustomer/graphql/getCustomer.graphql.d.ts:1]: 🤔 this is interesting...do you know why this changed?

File: scripts/initializers/order.js [hunk 5]
```diff
@@ -19,7 +20,9 @@ initializeDropin(async () => {
   const isTokenProvided = orderRef && orderRef.length > 20;
 
   // Handle redirects for user details pages
-  if (pathname === ORDER_DETAILS_PATH || pathname === CUSTOMER_ORDER_DETAILS_PATH) {
+  if (pathname === ORDER_DETAILS_PATH
+    || pathname === CUSTOMER_ORDER_DETAILS_PATH
+    || pathname === CUSTOMER_RETURN_DETAILS_PATH) {
```
Comment (sirugh) [inline scripts/initializers/order.js:25]: From what I can tell, the `handleUserOrdersRedirects` function seems to not be written for the `CUSTOMER_RETURN_DETAILS_PATH` handling. Do you need to update that function?

File: blocks/commerce-return-header/commerce-return-header.js (hunk 3 above)
Comment (Ferri0) [reply blocks/commerce-return-header/commerce-return-header.js:21]: Fixed ✅

File: scripts/initializers/order.js (hunk 5 above)
Comment (Ferri0) [reply scripts/initializers/order.js:25]: Nope, it's fine. Order details and return details pages will use same set of containers, and will be initialised in the same way, using order initializer. Return functionality is a part of order, and related initialisation and routing logic would be shared.

**Pull request merged**